### ISSUE TITLE
ask permissions _before_ adding default extensions as double click handlers

### ISSUE THREAD

**Issue by jessfraz** [KittyCAD/modeling-app#3007]
On windows when we install the app right now we are using tauri to define file types we can open. For the non-kcl file types (stl/obj/etc) we need to ask permission first or we come across as rude jsut doing it

we likely need to patch tauri for this.

**Comment by pierremtb** [comment]
Been looking to see if this was part of their config stuff at https://v2.tauri.app/reference/config/#wixconfig but it's not. I believe this is how other apps like VS Code do it right? Making you tick a box if you want the app to open those files?

Would assume that wix supports it under the hood, would have to dive in more to see how we could plug that in

**Comment by franknoirot** [comment]
Is this issue still relevant post-Electron migration @jessfraz and @pierremtb? I'd imagine it could be. Should we include a checkbox in our installer to toggle whether the additional file type associations get added?

**Comment by pierremtb** [comment]
@jessfraz I believe it is yeah. Good installers have checkboxes that you can tick for the file associations you want to have. What Jess was saying is that STL, STEP, and others should be optional, as you might want to keep say PrusaSlicer for STL. Let me check what electron-builder provides. It might need to be added to `scripts/installer.nsh`

**Comment by pierremtb** [comment]
Ok there has to be a way to inject stuff in the `registerFileAssociations` macro in a way that's opt-in or opt-out based on checkboxes, see https://github.com/electron-userland/electron-builder/blob/7f6c3fea6fea8cffa00a43413f5335097aca94b0/packages/app-builder-lib/src/targets/nsis/NsisTarget.ts#L734

But it's not exposed and could be somewhat of a big project, that might not provide that much value in the end? It's still possible right now to right-click > Open with... and select Zoo Modeling App (which triggers an import), that I think is a common pattern across apps any way? My take on this is it shouldn't be a must-have for v1 as long as we can right-click > Open with... > Zoo Studio and the import works. Thoughts @franknoirot @jessfraz?

**Comment by franknoirot** [comment]
I agree @pierremtb. I believe we currently make those secondary file associations like STL for the user on install. Are you suggesting we drop those for now as well?

**Comment by pierremtb** [comment]
@franknoirot I don't think so? At least the config doesn't say so https://github.com/KittyCAD/modeling-app/blob/4a6e3bf7216f83e035ec9161f43a528ced2fcc0d/electron-builder.yml#L40-L45

KCL is the only one there and I think that's the proper way for now. But open to disagreements of course

**Comment by franknoirot** [comment]
Oh dope no I am very likely wrong

**Comment by pierremtb** [comment]
@franknoirot I think the confusion comes from the tauri days, during which we applied it to lots of 3rd party cad formats, see https://github.com/KittyCAD/modeling-app/blob/tauri/src-tauri/tauri.release.conf.json#L9-L47

**Comment by franknoirot** [comment]
That seems sufficiently different of a topic to close this. We don't have an installer step on every platform, so we'd have to make a new pattern for something that is better accessed by users through the conventional "Open with" OS context menu. In the future if we have an installer window step, or we have a welcome screen on first ever app open for some other more important reason, we can discuss adding an opt-in checkbox there.
